### DISCUSSION TITLE
Not able to complete OAuth2 connection with Google Search Console: "Mint.TransportError"

### DISCUSSION THREAD

**Discussion by serg3k** [plausible/analytics#3020] [Self-Hosted Support] [answered]
Hello,

Please find my bug report below:

**Environment**

I'm hosting plausible behind caddy with let's encrypt SSL certificate.
Using docker-compose.yml from [this](https://github.com/plausible/hosting/commit/36d45da22818d67ec22c58f9bdb61d1d6497777d) commit, with `plausible-conf.env` properly configured.

**Problem**

Cannot connect Plausible to Google Search Console. Browser is showing 500, with stacktrace in logs (see below)

**Details**

When initiating connection to Google Search (clicking on blue "Continue with Google" on `/<domain>/settings/search-console` page button), I'm:
 - Being redirected to Google
 - Selecting my Google email/account
 - Confirming that I want to grant "Google Search Console"-related reader scopes to requesting client
 - Getting redirected back to `http://my-playsible-domain/auth/google/callback?..."
 - I see 500 and stacktrace in logs

At this point, plausible received `code` and wants to call `https://www.googleapis.com` to exchange code for an access_token.

This request fails with following Exception:
```
21:59:59.625 [error] #PID<0.3247.0> running PlausibleWeb.Endpoint (connection #PID<0.3246.0>, stream id 1) terminated
Server: pl.telepilot.co:80 (http)
Request: GET /auth/google/callback?state=%5B1,%22search-console%22%5D&code=4/0ACUTXXMU7dmSwXWqCWTO0qfErbE0eAFKQ_M2Eiq5AJr-ajYXNiDROt4yCwr_J8m8CCP5A&scope=email%20https://www.googleapis.com/auth/webmasters.readonly%20openid%20https://www.googleapis.com/auth/userinfo.email&authuser=0&prompt=consent
** (exit) an exception was raised:
    ** (MatchError) no match of right hand side value: {:error, %Mint.TransportError{reason: :nxdomain}}
        (plausible 0.0.1) lib/plausible/google/http.ex:128: Plausible.Google.HTTP.fetch_access_token/1
        (plausible 0.0.1) lib/plausible_web/controllers/auth_controller.ex:565: PlausibleWeb.AuthController.google_auth_callback/2
        (plausible 0.0.1) lib/plausible_web/controllers/auth_controller.ex:1: PlausibleWeb.AuthController.action/2
        (plausible 0.0.1) lib/plausible_web/controllers/auth_controller.ex:1: PlausibleWeb.AuthController.phoenix_controller_pipeline/2
        (phoenix 1.6.6) lib/phoenix/router.ex:355: Phoenix.Router.__call__/2
        (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint.plug_builder_call/2
        (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint."call (overridable 3)"/2
        (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint.call/2
```

As far as I can see, on [Line128](https://github.com/plausible/analytics/blob/b0c0214b6b70373f7a8c2a35ca6cf1a64b666a9a/lib/plausible/google/http.ex#L128) Plausible is trying to send POST request to `https://www.googleapis.com/oauth2/v4/token` and (as far as what I could google, I never worked with elixir before) is not able to resolve DNS name.. I guess it cannot resolve `www.googleapis.com`, right?

In order to verify that I went into my docker container:
> docker exec -it 1b222f301737 sh

And tried accessing `www.googleapis.com` -  DNS name is being resolved:

```
/tmp $ wget https://www.googleapis.com/
Connecting to www.googleapis.com (142.251.143.170:443)
wget: server returned error: HTTP/1.1 404 Not Found
/tmp $ wget https://analyticsreporting.googleapis.com
Connecting to analyticsreporting.googleapis.com (142.251.143.138:443)
wget: server returned error: HTTP/1.1 404 Not Found
/tmp $ wget https://www.googleapis.com//oauth2/v4/token
Connecting to www.googleapis.com (142.251.143.170:443)
wget: server returned error: HTTP/1.1 404 Not Found
/tmp $ wget https://www.googleapis.com/oauth2/v4/token
Connecting to www.googleapis.com (142.251.143.170:443)
wget: server returned error: HTTP/1.1 404 Not Found
/tmp $ curl -X POST https://www.googleapis.com/oauth2/v4/token
sh: curl: not found
```

**Question**

Am I missing something? What could the actual problem be in this case (if it's not DNS resolution).

I really appreciate your time you spend on reading and analysing this.

**Comment by serg3k** [comment, marked as answer]
I'm not sure what the real issue was. I was able to complete Google search Console integration after I added dns: 8.8.8.8 to docker-compose.yml:

```
  plausible:
    image: plausible/analytics
    restart: always
    command: sh -c "sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh run"
    depends_on:
      - plausible_db
      - plausible_events_db
      - mail
    ports:
     - 8000:8000
    **dns:
      - 8.8.8.8**
    networks:
      - plausible
      - caddy
    env_file:
      - plausible-conf.env
```

**Reply by ruslandoga** [reply]
@sergk8 thank you for documenting the solution!
